plugins/bufferlines/barbar.nix

```diff
@@ -69,249 +24,592 @@ let
-                        }
-                      );
-                    }) setIcons;
-                  in
-                  helpers.mkRaw (
-                    "{"
-                    + concatStringsSep "," (map (iconOption: "[${iconOption.key}] = ${iconOption.value}") setIconsList)
-                    + "}"
-                  );
-                filetype = with filetype; {
-                  custom_color = customColors;
-                  enabled = enable;
-                };
-                inherit separator;
-              };
+    no_name_title = helpers.defaultNullOpts.mkStr null ''
+      Sets the name of unnamed buffers.
 
-            handleStateOption =
-              stateOption:
-              with stateOption;
-              {
-                modified = handleBufferOption modified;
-                pinned = handleBufferOption pinned;
-              }
-              // (handleBufferOption (getAttrs (attrNames stateOption) stateOption));
-          in
-          (handleStateOption (getAttrs (attrNames stateOptions) cfg.icons))
-          // (genAttrs [
-            "alternate"
-            "current"
-            "inactive"
-            "visible"
-          ] (optionName: handleStateOption cfg.icons.${optionName}));
-        inherit (cfg) hide;
-        insert_at_end = cfg.insertAtEnd;
-        insert_at_start = cfg.insertAtStart;
-        maximum_padding = cfg.maximumPadding;
-        minimum_padding = cfg.minimumPadding;
-        maximum_length = cfg.maximumLength;
-        semantic_letters = cfg.semanticLetters;
-        inherit (cfg) letters;
-        no_name_title = cfg.noNameTitle;
-        sidebar_filetypes = cfg.sidebarFiletypes;
-      } // cfg.extraOptions;
-
-      keymaps = flatten (
-        mapAttrsToList (
-          optionName: funcName:
-          let
-            key = cfg.keymaps.${optionName};
-          in
-          optional (key != null) {
-            mode = "n";
-            inherit key;
-            action = "<Cmd>Buffer${funcName}<CR>";
-            options.silent = cfg.keymaps.silent;
-          }
-        ) keymapsActions
-      );
-    in
-    mkIf cfg.enable {
-      extraPlugins = with pkgs.vimPlugins; [
-        cfg.package
-        nvim-web-devicons
-      ];
+      By default format is `'[Buffer X]'` where `X` is the buffer number.
+      However, only a static string is accepted here.
+    '';
 
-      inherit keymaps;
+    semantic_letters = helpers.defaultNullOpts.mkBool true ''
+      If `true`, the letters for each buffer in buffer-pick mode will be assigned based on their name.
 
-      extraConfigLua = ''
-        require('barbar').setup(${helpers.toLuaObject setupOptions})
-      '';
+      Otherwise (or in case all letters are already assigned), the behavior is to assign letters in
+      the order of provided to `letters`.
+    '';
+
+    sidebar_filetypes = helpers.defaultNullOpts.mkAttrsOf (
+      with types;
+      either (enum [ true ]) (submodule {
```

Comment:
I think this'll be affected by #1618. I'm not implying you should change the option though.

Reply:
Yes, I don't see a better alternative anyway...